io/io/inc/TFile.h

```diff
@@ -181,6 +181,11 @@ class TFile : public TDirectoryFile {
 public:
    /// TFile status bits. BIT(13) is taken up by TObject
    enum EStatusBits {
+      // Produce files forward compatible with (unpatched) version older than
+      // v6.30 by recording the internal bits kIsOnHeap and kNotDeleted; Older
+      // releases were not explicitly setting those bits to the correct value
+      // but instead used verbatim the value stored in the file.
```

Comment:
Perhaps we can add to this comment that the `k630forwardCompatibility` value is used in `TObject`, i.e. when changing it here (however unlikely), we need to remember to adjust TObject streamer as well.

Reply:
```suggestion
      // but instead used verbatim the value stored in the file.
      // Note that to avoid a circular dependency, this value is used
      // hard coded in TObject.cxx.
```